### PR TITLE
ISSUE #2029: Add new model for new federal agency field

### LINKED ISSUE #2029: Create a new field Agency that exists on Domain Request and Domain Info
### Issue description

 #1793 and #1975 have the goal of making all agency values connected to the Agency table, instead of just the agency list that is hard encoded. However, we don't want to just automatically connect change the variable federal_agency to connect to the Federal Agency table, because then we will lose the values already entered (which are strings). We instead need to make a **new** field that holds a value that is connected to the agency table. After we have a field that exists for this on the domain request and domain information tables we will then be able to unblock the migration script ticket #1975

### Acceptance criteria

- [ ] add a new field to Domain Request and DomainInformation called agency that ties to the Federal Agency table
- [ ] superusers can see this but not analysts (this will just be empty for now so we don't need to show it to them and add confusion)
- [ ] displayed label on django admin should still say federal agency (hence, even more reason to not show for analysts, this may be confusing for analysts)

### Additional context

note, this should result in a migration file and permission change migration file, but as long as you can see this in django admin as a drop down as a superuser and add/remove it to Domains Information and domain request, you should not need a unit test. 

### Links to other issues

blocks #1975  (migration script to do next)
blocks #1793 (connect to drop down)

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #2029 

## Changes

* Adding in a new federal agency model which is pointing to the right table versus the "choices" that we are currently using

You can see the visual of the model changes in either Domain Information and/or Domain Request in Admin. Please note that there is a comment in the code mentioning that we'll be commenting it out afterwards bc it's hard/not worth our time to try to hide 1 field from the Analyst.

Screenshot 2024-04-17 at 4 30 19 PM

## Context for reviewers

* This is part 1 of 3
* Here's the [ticket 2](https://github.com/cisagov/manage.get.gov/issues/1975) and [ticket 3](https://github.com/cisagov/manage.get.gov/issues/1793) if you're curious what we're doing
#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/admin.py
Comment (rachidatecs) [inline src/registrar/admin.py:962]: Blocking: My understanding is we keep BOTH fields for now. Make the new field visible only for superadmins.

File: src/registrar/tests/test_admin.py
Comment (rachidatecs) [inline src/registrar/tests/test_admin.py:1789]: We'll keep both fields and make the new field visible only to superadmins

Comment (therealslimhsiehdy) [reply src/registrar/tests/test_admin.py:1861]: Yes we are keeping both fields for now, but we are going to comment out #updated_federal_agency after we get an approval so it doesn't show up for anyone on Admin since it's not useful

Comment (rachidatecs) [reply src/registrar/tests/test_admin.py:1861]: AC #2: superusers can see this but not analysts (this will just be empty for now so we don't need to show it to them and add confusion)

Comment (therealslimhsiehdy) [reply src/registrar/tests/test_admin.py:1861]: OH yes sorry this AC was not updated - Erin and I had a meeting with @abroddrick and agreed with that compromise! Apologies for the confusion!

Comment (rachidatecs) [reply src/registrar/tests/test_admin.py:1861]: Got it. APPROVED.

**Review by rachidatecs — APPROVED** [review]

**Pull request merged**
